kernel/kernel-api/src/main/java/io/delta/kernel/client/JsonHandler.java

```diff
@@ -98,4 +99,28 @@ CloseableIterator<ColumnarBatch> readJsonFiles(
         CloseableIterator<FileStatus> fileIter,
         StructType physicalSchema,
         Optional<Predicate> predicate) throws IOException;
+
+    /**
+     * Write each `Row` in given `data` serialized as JSON and written as separate line in
+     * destination file. This call either succeeds in creating the file with given contents or no
+     * file is created at all.
+     * <p>
+     * There are a few special cases that should be handled for specific data types:
+     * <ul>
+     *    <li><b>Date Type:</b> serialize as string of format <code>"yyyy-MM-dd"</code></li>
+     *    <li><b>Timestamp or Timestamp NTZ:</b> serialize as string of format
+     *    <code>"yyyy-MM-dd'T'HH:mm:ss.SSSXXX"</code></li>
```

Comment:
Yeah it seems the combining of these two scenarios is what's confusing since we have the non-JSON data types... Removing the other types here sounds good to me.

Reply:
Created: https://github.com/delta-io/delta/issues/2929